### DISCUSSION TITLE
how to configure bashls to analyze zsh filetypes

### DISCUSSION THREAD

**Discussion by vesiri** [LunarVim/LunarVim#4046] [Q&A]
im working with zsh shell and want to have an lsp enabled and am trying to use bashls for this.

i have tried updating the lsp settings using `:LspSettings bashls` and in the file bashls.json that this creates i have added this:

`{
  "bashIde.globPattern": "**/*@(.sh|.inc|.bash|.command|.zsh|zshrc|zsh_*|.zsh_*)"
}
`

However lsp status is still 'LS Inactive' when editing any zsh file.

How do I correctly configure bashls to also analyse zsh configs?

**Comment by salimp2009** [comment]
you can try this (if you want format shfmt to format it is available with :Mason)
```
local formatters = require "lvim.lsp.null-ls.formatters"
formatters.setup {
  { command = "shfmt", filetypes = { "sh", "zsh", "bash" } },
}

vim.filetype.add {
  extension = {
    zsh = "zsh",
  },
}

vim.list_extend(lvim.lsp.automatic_configuration.skipped_servers, { "bashls" })

local lsp_manager = require "lvim.lsp.manager"
lsp_manager.setup("bashls", {
  filetypes = { "sh", "zsh" },
  on_init = require("lvim.lsp").common_on_init,
  capabilities = require("lvim.lsp").common_capabilities(),
})

lsp_manager.setup("lua_ls", {
  filetypes = { "lua" },
  on_init = require("lvim.lsp").common_on_init,
  capabilities = require("lvim.lsp").common_capabilities(),
})
```

**Comment by ibootrc** [comment]
you can do config for nvim-lspconfig 

`
lspconfig.bashls.setup {
  filetypes = { 'sh', 'zsh', },
}
`
